README reorganise (#359)
UKGovernmentBEIS/beis-opss-psd: doc/deployment.md

## Deployment

Create your changes in a new branch and open a pull request.

> Pull requests trigger a deployment to the `int` space on GOV.UK PaaS of a temporary review application, which is then deleted when the PR is merged. The review application can be viewed at https://psd-pr-<PR NUMBER>-web.london.cloudapps.digital/. Ask someone on the team for the authentication credentials to access this.

Reviewer: ```suggestion
Pull requests trigger a deployment to the `int` space on GOV.UK PaaS of a temporary review application, which is then deleted when the PR is merged. The review application can be viewed at https://psd-pr-XXXX-web.london.cloudapps.digital/. Ask someone on the team for the authentication credentials to access this.
```
<PR NUMBER> does not display correctly as HTML.

```

* To configure Sentry (see the Sentry account section in [the root README](../README.md#sentry) to get these values):

```
    cf7 cups psd-sentry-env -p '{
        "SENTRY_DSN": "XXX",
        "SENTRY_CURRENT_ENV": "<<SPACE>>"
    }'
```

* To enable and add basic auth to the entire application (useful for deployment or non-production environments):

```
    cf7 cups psd-auth-env -p '{
        "BASIC_AUTH_USERNAME": "XXX",
        "BASIC_AUTH_PASSWORD": "XXX"
    }'
```

* To enable and add basic auth to the health check endpoint at `/health/all`:

```
    cf7 cups psd-health-env -p '{
        "HEALTH_CHECK_USERNAME": "XXX",
        "HEALTH_CHECK_PASSWORD": "XXX"
    }'
```

* To enable and add basic auth to the sidekiq monitoring UI at `/sidekiq`:

```
    cf7 cups psd-sidekiq-env -p '{
        "SIDEKIQ_USERNAME": "XXX",
        "SIDEKIQ_PASSWORD": "XXX"
    }'
```

* `psd-keycloak-env` should already be setup from [the keycloak steps](https://github.com/UKGovernmentBEIS/beis-opss/blob/master/keycloak/README.md#setup-clients).

Once all the credentials are created, the app can be deployed using:

    SPACE=<<space>> ./psd-web/deploy.sh

### GOV.UK Platform as a Service

> If you want to update any of the deployed instances, you'll need an account for [GOV.UK PaaS](https://admin.london.cloud.service.gov.uk/) - ask someone on the team to invite you.

Reviewer: ```suggestion
You'll need an account for [GOV.UK PaaS](https://admin.london.cloud.service.gov.uk/) to manage deployed instances.
```
Reader is not necessarily a member of the team.
